### PR TITLE
add integration test for subscriptons snd forks

### PR DESCRIPTION
**Pull request by jIrwinCline**
adds simple integration test for provideSubscriptions and createSubscription

### CODE REVIEW

File: tests/__tests__/sitka_module.test.ts [hunk 1]
```diff
@@ -1,98 +1,134 @@
...
+    expect(startingEditsState).toEqual(0)
+    // Validates that subscribed function is called and updates state
+    colorModule.handleColor("blue")
+    const updatedEditsState = textModule.getStateTestDelegate(store.getState()).numberOfEdits
+    expect(updatedEditsState).toEqual(1)
   })
 
   // FORKS
-  test('provideForks adds fork to Sitka', () => {
+  test('provideForks adds fork to Sitka', (done) => {
```
Comment (jIrwinCline) [inline tests/__tests__/sitka_module.test.ts:111]: This test, when switched to work with sitka factory shows an error referenceError: alert not defined

File: tests/__tests__/sitka_module.test.ts [hunk 2]
```diff
@@ -86,13 +89,39 @@ describe("SitkaModule", () => {
...
 
-//   // SUBSCRIPTION
-//   test('subscriptions are created/provided with provideSubscriptions & createSubscription', () => {
-//   })
+  // SUBSCRIPTION
+  test('subscriptions are created/provided with provideSubscriptions & createSubscription', () => {
+    const { sitka, store } = createSitkaAndStore()
+    const { text: textModule, color: colorModule } = sitka.getModules()
+    // Validates that we start with default state
+    const startingEditsState = textModule.getStateTestDelegate(store.getState()).numberOfEdits
```
Comment (rachoac) [inline tests/__tests__/sitka_module.test.ts:97]: no need to use the delegate to inspect store state, use store state. getStateTestDelegate should ONLY be used to test getState.

File: tests/__tests__/sitka_module.test.ts [hunk 3]
```diff
@@ -86,13 +89,39 @@ describe("SitkaModule", () => {
...
+    expect(startingEditsState).toEqual(0)
+    // Validates that subscribed function is called and updates state
+    colorModule.handleColor("blue")
+    const updatedEditsState = textModule.getStateTestDelegate(store.getState()).numberOfEdits
```
Comment (rachoac) [inline tests/__tests__/sitka_module.test.ts:101]: how do i know that handleColor blue isn't also changing the numberOfEdits using mergeState?

File: tests/__tests__/sitka_module.test.ts (hunk 2 above)
Comment (jIrwinCline) [reply tests/__tests__/sitka_module.test.ts]: removed except in the first instance where needed

File: tests/__tests__/sitka_module.test.ts (hunk 3 above)
Comment (jIrwinCline) [reply tests/__tests__/sitka_module.test.ts]: Okay good point instead I set up a spy

**Pull request merged**